SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -89,37 +89,63 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
-
 	//checkf(!SpatialOSInstance->IsConnected(), TEXT("SpatialOS should not be connected already. This is probably because we attempted to travel to a different level, which current isn't supported."));
 
 	// Set the timer manager.
 	TimerManager = &LoadedWorld->GetTimerManager();
 
-	// Connect to SpatialOS.
-	Connect();
-
 	// Set up manager objects.
 	EntityRegistry = NewObject<UEntityRegistry>(this);
-}
 
-void USpatialNetDriver::Connect()
-{
+	// Handle connection configurations.
+	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
+
 	if (Connection != nullptr)
 	{
 		return;
 	}
 
 	Connection = NewObject<USpatialWorkerConnection>();
+
+	bool bUseLocator = LoadedWorld->URL.HasOption(TEXT("spatialLocator"));
+
+	if (bUseLocator)
+	{
+		Connection->LocatorConfig.ProjectName = LoadedWorld->URL.GetOption(TEXT("project"), TEXT(""));
```

Comment:
I've just realized that this doesn't allow for specifying the deployment name. Looks like the GDK just defaults to the first active deployment in a project, which isn't desired behavior. I'm going to dig around and see if there's a quick solution to this.

Reply:
Added in a fix, although I haven't tested it with a project with multiple running deployments yet